### DISCUSSION TITLE
How to rename a camera without losing footage/events

### DISCUSSION THREAD

**Discussion by leccelecce** [blakeblackshear/frigate#9233] [Ask A Question] [answered]
I don't believe there is any official support for this currently.

Looks like a script would be required to:

- stop Frigate (to avoid clashes with background processes that touch cameras/recordings in the background)
- backup the sqlite DB
- edit the sqlite DB to rename the camera in all the tables where it's referenced
- -- this involves both updating the 'camera' field and the 'path' field
- walk the filesystem and rename the camera-named folders
- update the camera name in the config
- restart Frigate

I've not considered any downstream integrations e.g. HASS.

Anyone attempted this already?

[edit] Some SQL queries, accurate as of `0.13.0-rc1`

sqlite> update recordings set path=replace(path,'camera2','garden') where path like '%camera2%';
sqlite> update recordings set camera='garden' where camera = 'camera2';
sqlite> update event set camera='garden' where camera = 'camera2';
sqlite> update timeline set camera='garden' where camera = 'camera2';
sqlite> update regions set camera='garden' where camera = 'camera2';

**Comment by NickM-27** [comment, marked as answer]
yeah that pretty much covers it on the frigate side

**Reply by leccelecce** [reply]
So it turns out I missed something. You can't just rename the 'camera' column on the tables - there is a field called 'path' on the recordings table that also needs to be updated.

**Reply by maples3** [reply]
I tried doing this myself today, and you can't just update the paths without updating the locations on disk too.  I ended up getting things mixed up to the point where no old recordings would work, and revering the database wasn't an option because I moved the files on disk.  Fortunately I'm still experimenting with getting things set up, and 99% of the important settings are stored in the `config.yaml` file, so I just backed up my config.yaml, nuked the /config and /media directories, and started over.

If I were to do this again, I would try updating _only_ the database, _not_ move any directories on-disk.  I wouldn't touch any `path` columns (since I assume Frigate uses the stored path in the database instead of re-constructing it based on the camera name?), and see if that works.

Also, as of `0.15`, there are 2 more tables that have a `camera` column: `reviewsegment` and `previews`

**Reply by travis-fm** [reply]
I can confirm just updating the camera names in each table worked on my end. Didn't have to touch the paths or on-disk directories at all, and I was able to view footage and previews from several days ago no problem.  

Worst case, there _might_ be some empty directories for the old names once their retention period runs out.

**Reply by jmhunter** [reply]
To add to @travis-fm 's comments, I needed a couple of extra steps beyond simply the SQL updates:
* Obviously, rename the camera in config.yaml (in both the camera: and go2rtc: sections)
* Also update the names of the members of any camera_groups
* In my case I had a zone on one camera that was now named the same as a camera - and zones can't be called the same as a camera, so I had to rename the zone itself and also where it was referenced in required_zones
* I didn't recreate my frigate container, so the /tmp/cache directory within the container still contained files from the old camera name. I needed to remove these to avoid "Error occurred when attempting to maintain recording cache" errors:
```shell
docker compose exec -it frigate /bin/bash
  rm /tmp/cache/oldcameraname*.mp4
```
All seems to be working perfectly now :)
